fv/bpf_test.go

```diff
@@ -831,6 +833,140 @@ func describeBPFTests(opts ...bpfTestOpt) bool {
 					})
 				})
 
+				Describe("Test Load balancer service with external IP", func() {
+					srcIPRange := []string{}
+					externalIP := []string{extIP}
+					testSvcName := "test-lb-service-extip"
+					tgtPort := 8055
+					var testSvc *v1.Service
+					var ip []string
+					var port uint16
+					BeforeEach(func() {
+						testSvc = k8sCreateLBServiceWithEndPoints(k8sClient, testSvcName, "10.101.0.10", w[0][0], 80, tgtPort,
+							testOpts.protocol, externalIP, srcIPRange)
+						// when we point Load Balancer to a node in GCE it adds local routes to the external IP on the hosts.
+						// Similarity add local routes for externalIP on felixes[0], felixes[1]
+						felixes[1].Exec("ip", "route", "add", "local", extIP, "dev", "eth0")
+						felixes[0].Exec("ip", "route", "add", "local", extIP, "dev", "eth0")
+						ip = testSvc.Spec.ExternalIPs
+						port = uint16(testSvc.Spec.Ports[0].Port)
+					})
+
+					Context("Test LB-service with external IP and workloads", func() {
+
+						It("should have connectivity from workloads[1][0],[1][1] and [0][1] via external IP to workload 0", func() {
+							cc.ExpectSome(w[1][0], TargetIP(ip[0]), port)
+							cc.ExpectSome(w[1][1], TargetIP(ip[0]), port)
+							cc.ExpectSome(w[0][1], TargetIP(ip[0]), port)
+							cc.CheckConnectivity()
+						})
+					})
+
+					Context("Test LB-service with external IP and external Client", func() {
+						BeforeEach(func() {
```

Comment:
Think I'd set up the external client in the BeforeEach() above, just to save one extra nesting.  In fact, you could combine the workload test above with `cc.ExpectSome(externalClient, TargetIP(ip[0]), port)`.  FVs run so slowly, it's sometimes better to combine them instead of having super-precise tests.